src/app/[locale]/filter-set.tsx

```diff
@@ -10,14 +19,61 @@ interface Props {
   testId?: string;
 }
 
-export default function FilterSet({
-  title,
+type FilterOptionProps = Pick<
+  Props,
+  'title' | 'selectedFilters' | 'setSelectedFilters'
+> & {
+  searchResultFilter: SearchResultFilter;
+};
+
+type HeaderProps = Pick<Props, 'title' | 'searchResultFilters'> & {
+  setIsExpanded: Dispatch<React.SetStateAction<boolean>>;
+  setSearchText: Dispatch<React.SetStateAction<string>>;
+  searchText: string;
```

Comment:
We could harmonize the names of these properties?

https://github.com/colonial-heritage/dataset-browser/blob/main/src/app/%5Blocale%5D/client-filters.tsx#L38 uses `query` and `setQuery`.